tests/lock_test.py

```diff
@@ -59,9 +59,9 @@ def set_up():
     command = set_up_command(config_dir, logs_dir, work_dir, nginx_dir)
 
     dirs = [logs_dir, config_dir, work_dir]
-    # Travis and Circle CI set CI to true so we
-    # will always test Nginx's lock during CI
-    if os.environ.get('CI') == 'true' or util.exe_exists('nginx'):
+    # A specific environment variable for Azure Pipelines is SYSTEM_TEAMFOUNDATIONSERVERURI.
+    # If set, then we know we are on a CI job, and we test Nginx's lock.
+    if os.environ.get('SYSTEM_TEAMFOUNDATIONSERVERURI') == 'true' or util.exe_exists('nginx'):
```

Comment:
Is this environment variable's value "true"? Do we expect this environment variable to be set at this point or do we need to whitelist it in tox? Do we need to do anything special to have it exposed by Azure?

Reply:
So this variable is also used by codecov to detect an Azure environment. It is always set, and quite specific in my opinion to be sure that we are on Azure when it is set. Its value is `https://dev.azure.com/certbot/` here, and a valid test could be to check for a truthy value. And a config to tox needs to be done to pass the env variable properly to the script. I can fix all of that.

However, I would like to challenge the general approach, in an effort to not tie the code specifically to a CI provider.

The purpose of this control is unclear for me regarding checking we are on a CI. In my opinion, we could just drop this env variable control, and assume that nginx is installed so the test will be executed, because otherwise, the step installing nginx would have failed. 

Also in my opinion, having a logic inside the test to make it silently always succeed if nginx is not here feel wrong to me also, but changing that would require significant changes to the `tox` envs, which are not in the scope of this PR.

So, what do you think of simply remove the control to the environment variable here?

Reply:
I would like to keep the environment variable.

The reason we check `util.exe_exists('nginx')` is to allow this test to pass locally if nginx isn't installed. We do not want to allow this behavior in CI though so if the environment variable is set, we will try to run the tests with nginx.

What do you think about just changing this line back to `os.environ.get('CI') == 'true'` and setting this environment variable in our Azure config?

Reply:
Yes I understood the need to check for nginx at this time, for local execution of the tests. This would be useless with a refactoring of tox.ini, but it is out of scope.

However my point here is about the check for environment variable. I think it is useless, because the pipeline ensures that nginx is installed, so the tests will be executed, not silently skipped.

Reply:
I think it _should_ be useless, but the idea was to add an 2nd layer check and more robustness to help us ensure that if things go wrong now or in the future and nginx isn't found in CI, we don't unknowingly skip these tests. I personally still think there's some value in this especially since setting an environment variable is so easy.

However, it appears we weren't passing this variable through in `tox.ini` anyway so it was never set in Travis either. I still have the same preference, but with this in mind, you can do whatever you want here.

Reply:
After some discussion on Mattermost, we agreed that we can remove the check to the variable, and create an issue to also reorganize `tox.ini` to not need either the skipping when `nginx` binary is not present.

Issue to remove the nginx installation check is: https://github.com/certbot/certbot/issues/8121